apis/v1alpha1/opentelemetrycollector_types.go

```diff
@@ -365,6 +373,27 @@ type AutoscalerSpec struct {
 	TargetMemoryUtilization *int32 `json:"targetMemoryUtilization,omitempty"`
 }
 
+// MetricsConfigSpec defines a metrics config.
+type MetricsConfigSpec struct {
+	// CreateServiceMonitors specifies if ServiceMonitors should be created for the OpenTelemetry Collector.
+	// The operator.observability.prometheus feature gate must be enabled to use this feature.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Create ServiceMonitors for OpenTelemetry Collector"
+	CreateServiceMonitors bool `json:"createServiceMonitors,omitempty"`
```

Comment:
Is `CreateServiceMonitors` generic enough to handle the switch to the Collector when the time comes?  It would be nice if our Config API doesn't need any breaking changes when we can replace Prometheus with the Collector's exporting capabilities.

Reply:
Ok I agree the name is not as generic as it should be. How about `EnableMetrics`?

Reply:
I like that, and that will flex well into any metric generation solution.